src/plugins/func/azureFunc.ts

```diff
@@ -0,0 +1,109 @@
+import fs from "fs";
+import path from "path";
+import rimraf from "rimraf";
+import Serverless from "serverless";
+import { FuncPluginUtils } from "./funcUtils";
+
+export class AzureFuncPlugin {
+  public hooks: { [eventName: string]: Promise<any> };
+  public commands: any;
+  
+
+  public constructor(private serverless: Serverless, private options: Serverless.Options) {
+    this.hooks = {
+      "func:func": this.func.bind(this),
+      "func:add:add": this.add.bind(this),
+      "func:remove:remove": this.remove.bind(this)
+    };
+
+    this.commands = {
+      func: {
+        usage: "Add or remove functions",
+        lifecycleEvents: [
+          "func",
+        ],
+        commands: {
+          add: {
+            usage: "Add azure function",
+            lifecycleEvents: [
+              "add",
+            ],
+            options: {
+              name: {
+                usage: "Name of function to add",
+                shortcut: "n",
+              }
+            }
+          },
+          remove: {
+            usage: "Remove azure function",
+            lifecycleEvents: [
+              "remove",
+            ],
+            options: {
+              name: {
+                usage: "Name of function to remove",
+                shortcut: "n",
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private async func() {
+    this.serverless.cli.log("Use the func plugin to add or remove functions within Function App");
+  }
+
+  private async add() {
+    if (!("name" in this.options)) {
+      this.serverless.cli.log("Need to provide a name of function to add");
+      return;
```

Comment:
if you return instead of throwing error, would this exit other lifecycle?

Reply:
What other lifecycle are you referring to? This will just exit the process, and they would need to run again with a name specified

Reply:
ah you're right, I was thinking of other command with more lifecyles